src/test/java/htsjdk/samtools/cram/structure/BlockTest.java

```diff
@@ -0,0 +1,116 @@
+package htsjdk.samtools.cram.structure;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.CRAMException;
+import htsjdk.samtools.cram.common.CramVersions;
+import htsjdk.samtools.cram.common.Version;
+import htsjdk.samtools.cram.compression.ExternalCompressor;
+import htsjdk.samtools.cram.structure.block.*;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class BlockTest extends HtsjdkTest {
+
+    private void contentCheck(final Block actual,
+                             final byte[] expectedRaw,
+                             final byte[] expectedCompressed) {
+
+        // raw and compressed data are equal to what was given
+
+        Assert.assertEquals(actual.getUncompressedContent(), expectedRaw);
+        Assert.assertEquals(actual.getUncompressedContentSize(), expectedRaw.length);
+        Assert.assertEquals(actual.getCompressedContent(), expectedCompressed);
+        Assert.assertEquals(actual.getCompressedContentSize(), expectedCompressed.length);
+    }
+
+    private void rtCheck(final Block srcBlock, final byte[] testData, final Version[] versions) throws IOException {
```

Comment:
Part of the reason I suggested using a data provider was to eliminate this need for this kind of loop.